location/src/main/java/com/yas/location/service/CountryService.java

```diff
@@ -0,0 +1,56 @@
+package com.yas.location.service;
+
+import com.yas.location.exception.DuplicatedException;
+import com.yas.location.exception.NotFoundException;
+import com.yas.location.model.Country;
+import com.yas.location.utils.Constants;
+import com.yas.location.repository.CountryRepository;
+import com.yas.location.viewmodel.country.CountryPostVm;
+import com.yas.location.viewmodel.country.CountryVm;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@Transactional
+public class CountryService {
+    private final CountryRepository countryRepository;
+
+    public CountryService(CountryRepository countryRepository) {
+
+        this.countryRepository = countryRepository;
+    }
+
+    public Country create(CountryPostVm countryPostVm) {
+      validateExistedName(countryPostVm.name(), null);
```

Comment:
avoid passsing null downstream
should by using Jpa built in method 
```Java
countryRepository.existsByName(name)
````

Reply:
The current JPA version we are using does not support existsByName...
I am using JPQL to implement a custom query with exists semantics for both creating and updating cases.